contrib/keyboard-layout-per-window.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+
+# This script keeps track of active keyboard layouts per window.
+#
+# This script requires i3ipc-python package (install it from a system package
+# manager or pip).
+
+import i3ipc
+
+ipc = i3ipc.Connection()
+prev_focused = -1
+windows = {}
+
+def on_window_focus(ipc, event):
+    global windows, prev_focused
+
+    # Save current layout
+    layouts = {}
+    for input in ipc.get_inputs():
+        layouts[input.identifier] = input.xkb_active_layout_index
+    windows[prev_focused] = layouts
+
+    # Restore layout of the newly focused window
+    if event.container.id in windows:
```

Comment:
also better add `else` statement here switching to the default layout (index 0) if current window id not in `windows`

Reply:
Not a big win, I prefer simplicity